### ISSUE TITLE
Error highlighting is off in Windows

### ISSUE THREAD

**Issue by ceedubs** [unisonweb/unison#3657]
From Slack:

> I'm running ucm through WSL and Windows Terminal, and it looks like the coloring is wrong. I'm not sure if it's supposed to be highlighting different parts of the same words but it feels wrong
words but it feels wrong

<img width="601" alt="stuff" src="https://user-images.githubusercontent.com/977929/205366145-d337bc84-9011-46d6-9d0f-31825a31e1de.png">

**Comment by tankorsmash** [comment]
Here's a few more examples from:

    ucm version: 5f4b4f8 (built on 2022-11-30)

![image](https://user-images.githubusercontent.com/1312390/205366382-e5d44676-2e7e-4817-acf3-e02911c97a79.png)

![image](https://user-images.githubusercontent.com/1312390/205366409-3824e60d-0eba-4460-aca1-790b26c02349.png)

![image](https://user-images.githubusercontent.com/1312390/205366567-ccfe080e-d993-4623-a92d-dd4350a6957d.png)

**Comment by ChrisPenner** [comment]
> I'm running ucm through WSL and Windows Terminal

Sorry, for clarification, does this mean you're running WSL _inside of_ Windows Terminal when you see this?

Does this also occur if you run `ucm.exe` outside of WSL?

**Comment by tankorsmash** [comment]
tl;dr Yes, I can reproduce the incorrect indentation on `ucm.exe` outside of WSL.

---

I'm not sure exactly how it works internally, but I mean that instead of running Powershell or CMD directly, I'm using Windows Terminal, and then running Powershell 7.2 inside of that. I believe that means that instead of "Windows Console", I'm using "Windows Terminal".

So I am running WSL (Windows Subsystem for Linux), which is effectively Ubuntu through Windows Terminal, and then from within Ubuntu, running `ucm` the binary. 

> Does this also occur if you run `ucm.exe` outside of WSL?

I just downloaded the exe to try and reproduce (since I was running the binary through WSL before) so the version is different for `ucm.exe` (release/M3 (built on 2022-03-07)), but the output is mangled on Windows Console: 

![image](https://user-images.githubusercontent.com/1312390/207956095-dc548802-a968-4fb9-b0b1-bcf399ec7bde.png)

but running `ucm.exe` through Windows Terminal works out properly insofar as control codes:

![image](https://user-images.githubusercontent.com/1312390/207956150-f745b4c8-0766-4a78-b3f3-622c2350bf04.png)


but still has issues with coloring indentation:
![image](https://user-images.githubusercontent.com/1312390/207954771-656827f4-b2c5-42dc-bd34-9dac3525783d.png)


The code to repro that is:

```elm
name = "ASDAD"


someFunc txt =
	Text.join ?\n name
```
